Entity Web API generation

bohdan-harniuk opened the pull request:
**Description** (*)
This PR adds the possibility of generating REST API endpoints for managing entity generated with the entity creator feature.

**What was done:**
1. added to entity creator dialog checkbox that indicates if REST API endpoints should be created (by default is disabled)
2. generates API for the DeleteByIdCommand class, adds declaration to the webapi.xml file, replaces the original class type on the newly created interface whenever it is used
3. generates API for the SaveCommand class, adds declaration to the webapi.xml file, replaces the original class type on the newly created interface whenever it is used
4. generates API for the GetListQuery class, adds declaration to the webapi.xml file, replaces the original class type on the newly created interface whenever it is used
5. edited tests that related to the affected parts of the functionality

<img width="503" alt="Screenshot 2021-06-07 at 09 08 35" src="https://user-images.githubusercontent.com/31848341/120967360-07705900-c770-11eb-975a-738e4c6a80ca.png">
<img width="1074" alt="Screenshot 2021-06-07 at 09 18 54" src="https://user-images.githubusercontent.com/31848341/120968364-74382300-c771-11eb-920d-7ec4dd2e417b.png">
<img width="1074" alt="Screenshot 2021-06-07 at 09 18 00" src="https://user-images.githubusercontent.com/31848341/120968370-7601e680-c771-11eb-86b8-7a6e6edf4fd3.png">
<img width="1074" alt="Screenshot 2021-06-07 at 09 17 25" src="https://user-images.githubusercontent.com/31848341/120968372-77331380-c771-11eb-8ffc-2669d980b367.png">
<img width="1628" alt="Screenshot 2021-06-07 at 09 15 49" src="https://user-images.githubusercontent.com/31848341/120968375-77cbaa00-c771-11eb-9b19-66a2b22f3f69.png">

**Contribution checklist** (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with integration/functional tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

eduard13 commented on src/com/magento/idea/magento2plugin/actions/generation/data/DeleteEntityControllerFileData.java:
```diff
@@ -26,12 +27,14 @@ public DeleteEntityControllerFileData(
             final @NotNull String acl,
-            final @NotNull String entityId
+            final @NotNull String entityId,
+            final boolean isDeleteCommandHasInterface
```
Let's replace this one to a more readable version: `hasDeleteCommandInterface` or `deleteCommandHasInterface`.

```suggestion
            final boolean hasDeleteCommandInterface
```

bohdan-harniuk replied:
Done.

eduard13 commented on src/com/magento/idea/magento2plugin/actions/generation/data/SaveEntityControllerFileData.java:
```diff
@@ -35,7 +37,8 @@ public SaveEntityControllerFileData(
             final @NotNull String dtoInterfaceName,
-            final boolean isDtoWithInterface
+            final boolean isDtoWithInterface,
+            final boolean isSaveCommandHasInterface
```
Same here.

eduard13 commented on src/com/magento/idea/magento2plugin/actions/generation/data/UiComponentDataProviderData.java:
```diff
@@ -32,17 +33,20 @@ public UiComponentDataProviderData(
             final String entityName,
-            final String entityIdFieldName
+            final String entityIdFieldName,
+            final boolean isQueryHasInterface
```
Same here

bohdan-harniuk replied:
Done.

eduard13 reviewed (changes requested):
Hi @bohdan-harniuk, thank you for this cool improvement. Please check the below suggestions, and additionally I have a concern regarding the checkbox and what happens if you don't select the pointed checkboxes.

![image](https://user-images.githubusercontent.com/15868188/121127111-f4be5880-c831-11eb-9397-39b2dc9c82b2.png)

So, basically we should have them as dependency or make them individually, because the end-user may not need the Admin UiComponents, but needs only the WebAPI required files.

Please let me know if there are any questions.
Thank you.

bohdan-harniuk commented:
Hello, @eduard13! Thank you for the great ideas in your code review!

**What was done:**

1. fixed bug when creating entity without DTO interface (generating was stuck on the data mapper generator)
2. changed bool parameters names (I had to name methods as IsHas... because of AvoidFieldNameMatchingMethodName PMD rule)
3. added commands, query and data mapper generation when creating admin UI components checkbox is unchecked

Thank you!!! Now it looks better!

Could you please proceed with the code review?

eduard13 reviewed (approved):
Good job 👍
